goldenmaster/Plugins/ApiGear/Source/ApiGear/Public/ApiGearConnectionsStore.h

```diff
@@ -0,0 +1,65 @@
+// Copyright Epic Games, Inc. All Rights Reserved
+#pragma once
+
+#include "CoreMinimal.h"
+#include "UObject/ObjectMacros.h"
+#include "UObject/Object.h"
+#include "Subsystems/EngineSubsystem.h"
+#include "Engine/EngineTypes.h"
+#include "Templates/SharedPointer.h"
+#include "ApiGearConnection.h"
+#include "ApiGearConnectionsStore.generated.h"
+
+/**
+ * Implements the connections store for the ApiGear plugin.
+ */
```

Comment:
The class has two things - one is managing current connections - this is connection map and the other is "managing" settings on startup on tear down (saving) - connected with factories.
The problem with this class is that it allows adding any connection - not olink only - it can be different plugin, then it allows saving it, but it won't be created on startup as it only has the factory for olink. It even allows adding the factory, but it won't be saved.
The workflow with it should be explained in the comment. I don't have idea how to solve that. Maybe you should not be able the connection, only the factory? (but it doesn't solve the save issue)
maybe the class like that uses the connection type - here the olink adapters should check if factory is registered and register it? or when registering a connection you should also point an object that is able to create it?

Reply:
I moved the registration of the olink factory to a different module. That should also work the same way for modules in another plugin.
They register their factories on module startup, before this class is initialized.
I added a short description how it works and where to look for an example